holoviews/element/graphs.py

```diff
@@ -0,0 +1,259 @@
+import param
+import numpy as np
+
+from ..core import Dimension, Dataset, Element2D
+from ..core.dimension import redim
+from ..core.util import max_range
+from ..core.operation import Operation
+from .chart import Points
+from .path import Path
+
+class graph_redim(redim):
+    """
+    Extension for the redim utility that allows re-dimensioning
+    Graph objects including their nodes and edgepaths.
+    """
+
+    def __call__(self, specs=None, **dimensions):
+        redimmed = super(graph_redim, self).__call__(specs, **dimensions)
+        new_data = (redimmed.data,)
+        if self.parent.nodes:
+            new_data = new_data + (self.parent.nodes.redim(specs, **dimensions),)
+        if self.parent._edgepaths:
+            new_data = new_data + (self.parent.edgepaths.redim(specs, **dimensions),)
+        return redimmed.clone(new_data)
+
+
+def circular_layout(nodes):
+    N = len(nodes)
+    circ = np.pi/N*np.arange(N)*2
+    x = np.cos(circ)
+    y = np.sin(circ)
+    return (x, y, nodes)
+
+
+class layout_nodes(Operation):
+    """
+    Accepts a Graph and lays out the corresponding nodes with the
+    supplied networkx layout function. If no layout function is
+    supplied uses a simple circular_layout function.
+    """
```

Comment:
It's fine for this operation to live here for now. But if we end up with a bunch more operations, they should move to that module.